crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -0,0 +1,269 @@
+                    {
+                        info!("Log request received: {request:?}");
+                        self.config
+                            .current_operations
+                            .insert(message.topic.name.clone());
+                        self.handle_logfile_request_operation(&message.topic, &request)
+                            .await?;
+                    }
+                }
+                Err(err) => {
+                    error!("Incorrect log request payload: {}", err);
+                }
+            }
+        } else {
+            error!(
+                "Received unexpected message on topic: {}",
+                message.topic.name
+            );
+        }
+
+        Ok(())
+    }
+
+    pub async fn handle_logfile_request_operation(
+        &mut self,
+        topic: &Topic,
+        request: &LogRequestPayload,
+    ) -> Result<(), LogManagementError> {
+        match self.execute_logfile_request_operation(topic, request).await {
+            Ok(()) => Ok(()),
+            Err(error) => {
+                let error_message = format!("Handling of operation failed with {}", error);
+                self.mqtt_publisher
+                    .send(create_operation_message_with_reason(
+                        topic,
+                        request,
+                        CommandStatus::Failed,
+                        &error_message,
+                    ))
+                    .await?;
+                self.config.current_operations.remove(&topic.name);
+                error!("{}", error_message);
+                Ok(())
+            }
+        }
+    }
+
+    /// executes the log file request
+    ///
+    /// - sends request executing (mqtt)
+    /// - uploads log content (http)
+    /// - sends request successful (mqtt)
+    async fn execute_logfile_request_operation(
+        &mut self,
+        topic: &Topic,
+        request: &LogRequestPayload,
+    ) -> Result<(), LogManagementError> {
+        if !request.status.eq(&CommandStatus::Executing) {
+            self.mqtt_publisher
+                .send(create_operation_message(
+                    topic,
+                    request,
+                    CommandStatus::Executing,
+                ))
+                .await?;
+        }
+
+        let log_content = log_manager::new_read_logs(
+            &self.plugin_config.files,
+            &request.log.log_type,
+            request.log.date_from,
+            request.log.lines,
+            &request.log.search_text,
+        )?;
+
+        self.send_log_file_http(log_content, request.tedge_url.clone())
+            .await?;
+
+        self.mqtt_publisher
+            .send(create_operation_message(
+                topic,
+                request,
+                CommandStatus::Successful,
+            ))
+            .await?;
+
+        self.config.current_operations.remove(&topic.name);
+        info!("Log request processed.");
```

Comment:
It's good to have a log message here, but if it's without precise information like type and command ID, the log message has very limited value in debugging. I'd add at least the target log type inside the log message.